authLdap.php

```diff
@@ -412,33 +414,32 @@ function authLdap_get_uid($username)
 }
 
 /**
- * Get the user's current role
+ * Get the user's current roles
  *
  * Returns empty string if not found.
  *
  * @param int $uid wordpress user id
- * @return string role, empty if none found
+ * @return array roles, empty if none found
  */
-function authLdap_user_role($uid)
+function authLdap_user_roles($uid)
 {
     global $wpdb;
 
     if (!$uid) {
-        return '';
+        return array( );
```

Comment:
I'm afraid I'm stuck on a legacy RH6 box with PHP 5.3.  Since array( ) is supported going forward, why not use it?  There are other places in your code that you use array( ), by the way.  And this plugin works in my setup.